replacy/__init__.py

```diff
@@ -56,19 +80,21 @@ class ReplaceMatcher:
     validate_match_dict = validate_match_dict
 
     def __init__(
-        self,
-        nlp,
-        match_dict=None,
-        forms_lookup=None,
-        custom_match_hooks: Optional[ModuleType] = None,
-        allow_multiple_whitespaces=False,
-        max_suggestions_count=1000,
-        lm_path=None,
-        filter_suggestions=False,
-        default_max_count=None,
-        debug=False,
+            self,
+            nlp,
+            match_dict=None,
+            forms_lookup=None,
+            custom_match_hooks: Optional[ModuleType] = None,
+            allow_multiple_whitespaces=False,
+            max_suggestions_count=1000,
+            lm_path=None,
+            filter_suggestions=False,
+            default_max_count=None,
+            debug=False,
+            extended_span=False
```

Comment:
is it possible to have this be `SpanClass` or something and instead of being a flag, we pass in the span class we want to use? and it can default to `spacy.tokens.Span`, but if needed, we pass in `ESpan`? It would need to implement a few things differently, but I think it is doable?